lib/prestashop/mapper/extension.rb

```diff
@@ -40,11 +40,7 @@ def find_by options = {}
         #   Car.all(display: ['name']) # => [{ name: { language: { attr: { id: 2, href: 'http://localhost.com/api/languages/2'}, val: 'BMW 7'} }]
         #
         def all options = {}
-          result = if options[:display] 
-            Client.read self.resource, nil, display: options[:display]
-          else
-            Client.read self.resource
-          end
+          result = Client.read self.resource, nil, options
```

Comment:
Why did you removed the condition?

Reply:
:) I was expecting your question.

Here's my need : pass `:price`  option in  `options` hash with `:display` option to `Client.read`

In previous code it was impossible. Only `:display` was allowed. So two solutions :

1.  filtering options to keep only `:price` and  `:display` options

1. or be more permissive to pass whole options to `Client.read` so the conditions wasn't useful because by default second arg is nil and  third arg is empty hash and results to one single instruction `Client.read self.resource, nil, options`

I checked the code and I have not found anything against the second solution.

Did I broke something ?